File: app/src/main/java/com/ntech/fourtop/adapters/ImageAdapter.java
package com.ntech.fourtop.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ntech.fourtop.databinding.ItemImageBinding;
import com.ntech.fourtop.network.pojo.DataImage;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder> {

    private List<DataImage> images;

    public ImageAdapter() {
        images = new ArrayList<>();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ItemImageBinding
                .inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(images.get(position));
    }

    @Override
    public int getItemCount() {
        return images.size();
    }

    public void setImages(List<DataImage> images) {
        this.images = images;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private ItemImageBinding binding;

        public ViewHolder(ItemImageBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(DataImage image) {
            String textRating = "Рейтинг: " + image.getRate();
            binding.textRating.setText(textRating);
            Picasso.get().load(image.getLink()).into(binding.imageItem);
        }
    }
}

File: app/src/main/java/com/emikhalets/voteapp/adapters/ImageAdapter.kt
package com.emikhalets.voteapp.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.emikhalets.voteapp.databinding.ItemImageBinding
import com.emikhalets.voteapp.network.pojo.DataImage
import com.squareup.picasso.Picasso
import java.util.*

class ImageAdapter : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {
    private var images: List<DataImage>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(ItemImageBinding
                .inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(images[position])
    }

    override fun getItemCount(): Int {
        return images.size
    }

    fun setImages(images: List<DataImage>) {
        this.images = images
        notifyDataSetChanged()
    }

    inner class ViewHolder(private val binding: ItemImageBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(image: DataImage) {
            val textRating = "Рейтинг: " + image.rate
            binding.textRating.text = textRating
            Picasso.get().load(image.link).into(binding.imageItem)
        }
    }

    init {
        images = ArrayList()
    }
}